### PR TITLE
Api v3

### PR DESCRIPTION
**Pull request by escopecz**
https://mautic.atlassian.net/browse/TPROD-346

These changes are needed for https://github.com/mautic/mautic/pull/11724

This PR removes endpoints that the Mautic project do not use as we have no resources to maintain all the endpoints. Support for other endpoints can be added any time. As this is a major release and can contain breaking changes I decided to go with this approach as the API v2 will stop working in few days.

I was developing with `tests/Functional/LiveTest.php` which makes all the queries we'll need to make sure it's working correctly. Then there are bunch of unit tests that can run in the CI.

This PR is also moving the scattered docs into `Readme.md` with useful examples.

The changes here can be tested with a live Transifex project where you are a maintainer and this project has a `cs` translation. DO NOT TEST WITH A PRODUCTION PROJECT. More details in the [readme.md](https://github.com/mautic/Transifex-API/pull/4/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)

### CODE REVIEW

**Comment by RCheesley** [comment]
Is there any way to test this? Guessing maybe not?

**Comment by escopecz** [comment]
It's not yet done. I'm currently discussing whit Transifex support why one endpoint works for me and another gives me 403 permission denied response. So that's currently a blocker but it will be quite a refactoring. It will be a BC break obviously so I also updated dependencies and PHP versions starting with 7.4 so we could support M4 too.

**Comment by RCheesley** [comment]
Ah OK! Good luck with getting to the bottom of it all!

I guess if it's a BC break we have to wait for Mautic 5 - does this mean that languages will potentially stop working for Mautic 4 until we get that released?

**Comment by escopecz** [comment]
I meant that it will be a BC break for this repository/library so we'll have to release it as a major version. We can implement it for M4. It would break some plugins using this library but they would stop working as the old API is getting discontinued anyway so upgrading it to a working API is not as bad.

**Comment by RCheesley** [comment]
Ah OK that makes sense, thanks for the clarification!

**Review by sdoering — COMMENTED** [review]
Tested the PR in Gitpod by following the helpful hints from the documentation and by @escopecz . Thanks for helping me doing my first PR testing (ever) on an OS project. 

Steps I followed: 

1. Open the PR in gitpod
2. composer install dependencies
3. run `composer test` with the following result: 

```
gitpod /workspace/Transifex-API (api-v3) $ composer test
> vendor/bin/phpunit
PHPUnit 9.5.26 by Sebastian Bergmann and contributors.

.....................                                             21 / 21 (100%)

Time: 00:04.526, Memory: 6.00 MB

OK (21 tests, 94 assertions)
```
4. Did separate steps for functional and unittests, just to be sure: 
```
gitpod /workspace/Transifex-API (api-v3) $ composer test -- --testsuite=Functional
> vendor/bin/phpunit '--testsuite=Functional'
PHPUnit 9.5.26 by Sebastian Bergmann and contributors.

.                                                                   1 / 1 (100%)

Time: 00:04.539, Memory: 6.00 MB

OK (1 test, 32 assertions)
```
```
gitpod /workspace/Transifex-API (api-v3) $ composer test -- --testsuite=Unit
> vendor/bin/phpunit '--testsuite=Unit'
PHPUnit 9.5.26 by Sebastian Bergmann and contributors.

....................                                              20 / 20 (100%)

Time: 00:00.009, Memory: 6.00 MB

OK (20 tests, 62 assertions)
```
5. Ran codestyle checks: 
```
gitpod /workspace/Transifex-API (api-v3) $ composer cs
> vendor/bin/php-cs-fixer fix -v --dry-run --diff
PHP CS Fixer 3.13.0 Oliva by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.1.12
Loaded config default from "/workspace/Transifex-API/.php-cs-fixer.php".
Using cache file ".php-cs-fixer.cache".
SSSSSSSSSSSSSSSSSSSSSSSSSS                                                                                                                             26 / 26 (100%)
Legend: .-no changes, F-fixed, S-skipped (cached or empty file), I-invalid file syntax (file ignored), E-error

Checked all files in 0.003 seconds, 14.000 MB memory used

Detected deprecations in use:
- Rule "no_trailing_comma_in_singleline_array" is deprecated. Use "no_trailing_comma_in_singleline" instead.
```
6. Ran static analysis checks:
```
gitpod /workspace/Transifex-API (api-v3) $ composer phpstan
> vendor/bin/phpstan analyse src tests -l 5
 26/26 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

                                                                                                                        
 [OK] No errors                                                                                                         
                                                                                                                        
```

All checks ran 100% fine - thanks again for enabling me to support Mautic. 

Best regards, 
Sven

File: README.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    function (ResponseInterface $response, Promise $promise) {
+        echo "Translation for {$promise->getFilePath()} was uploaded successfully";
+    },
+    function (ResponseException $exception, Promise $promise) {
+        echo "Translation upload for {$promise->getFilePath()} failed with {$exception->getMessage()}";
+    }
+);
+```
+
+#### Download a traslation
```
Comment (tejasync) [inline README.md]:
```suggestion
#### Download a translation
```
Nitpicking.

**Review by tejasync — APPROVED** [review]
The code look good to me! 🎉  
Nicely done, thank you @escopecz!👌🏻

**Pull request merged**